tests/goss-linux.yaml

```diff
@@ -17,6 +17,10 @@ command:
   datadog-agent:
     exec: datadog-agent version
     exit-status: 0
+  default_java:
+    exec: java --version
+    exit-status:
+      not: 0
```

Comment:
This will fail as you removed any traces of `java` in the default `PATH`

Reply:
So should we remove this condition for the test harness? This was included in the test harness that was successful.

Reply:
- what is the intent of this PR?
- Is this change aligned with the intent of this PR?

Reply:
removing default jdk, the condition passes if there isn't a default jdk that's picked for linux builds. 
My assumption is that we kept the condition in to see if our changes worked, but now that it does work I'm not sure if we should include this in the PR. Correct me if I'm wrong.